### PR TITLE
Expose case-insensitive variants of `_starts_with` and `_ends_with`

### LINKED ISSUE #18702: Cannot use _istarts_with and _iends_with filters
### Describe the Bug

I've just discovered directus has support for some case insensitive filters, added with https://github.com/directus/directus/pull/11737 but I cannot find a way to use _istarts_with and _iends_with operators.

### To Reproduce

- Create a collection with a string field.
- Call the items API with `filter={"myField":{"_istarts_with":"foo"}}`

The API returns the following error:
```
{
	"errors": [
		{
			"message": "\"string\" field type does not contain the \"_istarts_with\" filter operator",
			"extensions": {
				"code": "INVALID_QUERY"
			}
		}
	]
}
```

### Directus Version

v10.1.1

### Hosting Strategy

Self-Hosted (Docker Image)

### PR DESCRIPTION
**Pull request by gitstart**
Fixes: #18702

 To see the changes working you need to run `pnpm build` in the `packages/utils` and `packages/types` folders.
- Create a collection and a string field (First button image below)
- Create data using the interface and query the data using the API.

This is the query format, replace whatever is between parentheses (including the parentheses themselves) with the corresponding data:

`http://localhost:8055/items/(collectionName)?filter={"(fieldName)":{"(filterName)":"(text)"}}`

Filters:
`_iends_with`
`_niends_with`
`_istarts_with`
`_nistarts_with`

You need to login to access the API, follow these [steps](https://docs.directus.io/reference/authentication.html#login)

Then just add the token as described [here](https://docs.directus.io/reference/authentication.html#access-tokens)

Tokens expire fast so to increase the time add this to the `.env`, and restart the server
`ACCESS_TOKEN_TTL="1y"`

https://www.loom.com/share/f1198ce2a697494da3758a9562d69e34?sid=cea79eeb-aadf-48f4-b5f3-3eb5413627fd

___
This code was written and reviewed by GitStart Community. Growing great engineers, one PR at a time.

### CODE REVIEW

File: .changeset/dirty-starfishes-dress.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@directus/types': patch
+'@directus/utils': patch
+---
+
+Add support for `_istarts_with` and `_iends_with` filters
```
Comment (paescuj) [inline .changeset/dirty-starfishes-dress.md:6]:
```suggestion
Expose case-insensitive variants of `_starts_with` and `_ends_with` filter operators
```

**Review by paescuj — APPROVED** [review]
Thanks!

**Review by jaads — APPROVED** [review]
LGTM! Nice additions @paescuj 🎉

**Comment by gitstart** [comment]
Thanks @paescuj @jaads 🙏

**Pull request merged**